src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -176,3 +181,94 @@ QVariant QgsPointCloudDataProvider::metadataClassStatistic( const QString &, con
 {
   return QVariant();
 }
+
+QVector<QMap<QString, QVariant>> QgsPointCloudDataProvider::identify(
+                                QgsPointCloudLayer *layer,
+                                float maximumError,
+                                float rootErrorPixels,
+                                QgsGeometry extentGeometry,
+                                const QgsDoubleRange extentZRange )
+{
+  QVector<QMap<QString, QVariant>> acceptedPoints;
+
+  QgsPointCloudIndex *index = this->index();
+  const IndexedPointCloudNode root = index->root();
+
+
+  QgsPointCloudLayerElevationProperties *properties = dynamic_cast<QgsPointCloudLayerElevationProperties *>( layer->elevationProperties() );
```

Comment:
I think we should remove the dependance on QgsPointCloudLayer here, and require that callers manually handle the elevation properties related logic themselves if desired.